specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-01-28-preview/HybridCompute.json

```diff
@@ -0,0 +1,1205 @@
+        "error": {
+          "description": "The error details.",
+          "$ref": "#/definitions/ErrorDetail"
+        }
+      },
+      "required": [
+        "error"
+      ]
+    },
+    "Identity": {
+      "title": "Managed Identity.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "The identity type.",
+          "type": "string"
+        },
+        "principalId": {
+          "readOnly": true,
+          "description": "The identity's principal id.",
+          "type": "string"
+        },
+        "tenantId": {
+          "readOnly": true,
+          "description": "The identity's tenant id.",
+          "type": "string"
+        }
+      }
+    },
+    "MachineExtension": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "allOf": [
+            {
+              "$ref": "#/definitions/MachineExtensionProperties"
+            }
+          ],
+          "description": "Describes Machine Extension Properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "Describes a Machine Extension."
+    },
+    "MachineExtensionUpdate": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "allOf": [
+            {
+              "$ref": "#/definitions/MachineExtensionUpdateProperties"
+            }
+          ],
+          "description": "Describes Machine Extension Update Properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/UpdateResource"
+        }
+      ],
+      "description": "Describes a Machine Extension Update."
+    },
+    "MachineExtensionProperties": {
+      "properties": {
+        "forceUpdateTag": {
+          "type": "string",
+          "description": "How the extension handler should be forced to update even if the extension configuration has not changed."
+        },
+        "publisher": {
+          "type": "string",
+          "description": "The name of the extension handler publisher."
+        },
+        "type": {
+          "type": "string",
+          "description": "Specifies the type of the extension; an example is \"CustomScriptExtension\"."
+        },
+        "typeHandlerVersion": {
+          "type": "string",
+          "description": "Specifies the version of the script handler."
+        },
+        "autoUpgradeMinorVersion": {
+          "type": "boolean",
+          "description": "Indicates whether the extension should use a newer minor version if one is available at deployment time. Once deployed, however, the extension will not upgrade minor versions unless redeployed, even with this property set to true."
+        },
+        "settings": {
+          "type": "object",
+          "description": "Json formatted public settings for the extension."
```

Comment:
This is under-specification of the settings.  In many languages, there will be no way to serialize or deserialize this value.  You need to have some modelling of these settings

Reply:
@markcowl  - Settings is an open type that is used by Extension publishers to include properties specific to their Extension. Do you a suggestion on how to better model this?